### PR TITLE
CTM-ETM coupling: generic transformation nodes

### PR DESCRIPTION
**Pull request by noracato**
This PR allows for circularities in the graph via transformation nodes, and solves these circularities by locally setting net-shares and net-demands on the transformation nodes.

Algorithm of nettifying will be improved in the future.

Dependent on quintel/atlas#170, quintel/refinery#62

### CODE REVIEW

File: app/models/qernel/reconciliation/helper.rb [hunk 1]
```diff
@@ -28,6 +28,8 @@ def supply_demand_balance(plugin)
           .installed_adapters_of_type(:producer)
           .sum(&:carrier_demand)
 
+        #  TODO: add transformation here
```
Comment (mabijkerk) [inline app/models/qernel/reconciliation/helper.rb:31]: @noracato there's an open to do here. Should it be addressed?

File: app/models/qernel/merit_facade/transformation_adapter.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          number_of_units: target_api.number_of_units
+        )
+      end
+
+      def installed?
+        producer? || consumer?
+      end
+
+      # Does not inject demand, as this node is expected to be a preset demand node
+      # TODO: check if it should inject for carrier specfic
```
Comment (noracato) [inline app/models/qernel/merit_facade/transformation_adapter.rb:55]: This one should also go indeed

File: app/models/qernel/reconciliation/helper.rb (hunk 1 above)
Comment (noracato) [reply app/models/qernel/reconciliation/helper.rb:31]: Yep

File: app/serializers/reconciliation_csv_serializer.rb [hunk 3]
```diff
@@ -5,14 +5,14 @@ class ReconciliationCSVSerializer < CausalityCurvesCSVSerializer
   private
 
   def producer_types
-    %i[producer import storage]
+    %i[producer flex]
   end
 
   def consumer_types
-    %i[consumer export storage]
+    %i[consumer flex]
```
Comment (mabijkerk) [inline app/serializers/reconciliation_csv_serializer.rb:12]: Why are export and storage removed here? These attributes still exist on ETSource for the natural gas nodes. For example `energy_export_natural_gas_backup` has the following:
```
- network_gas.profile = network_gas_export
```

**Review by louispt1 — COMMENTED** [review]
Looks good, just spotted one TODO other the one Mathijs caught.

**Comment by mabijkerk** [comment]
See https://github.com/quintel/etsource/pull/3107 for more information.

File: app/serializers/reconciliation_csv_serializer.rb (hunk 3 above)
Comment (noracato) [reply app/serializers/reconciliation_csv_serializer.rb:12]: @kndehaan can you maybe check this question, as you were the one working on this?

Comment (kndehaan) [reply app/serializers/reconciliation_csv_serializer.rb]: I thought that this class was only used for the hydrogen csv, but now see that it's also used for the network gas csv, my bad. Changes reverted and class description updated with https://github.com/quintel/etengine/pull/1455/commits/be650f4e7da43326860f92cad370b430759c90bf.

**Pull request merged**